### PR TITLE
refactor(sort-mixin): Include BaseDropdown component into Vue 3 migration test package

### PR DESCRIPTION
**Pull request by alvarodE**
## Motivation and context

Include BaseDropdown component into Vue 3 migration test package solving the issues found with VueUse.

Also, animation components' warnings are solved in order to simplify debugging.

- [x] Open issue: [EMP-3937](https://searchbroker.atlassian.net/browse/EMP-3937)

## What is the destination branch of this PR?

- [x] `Main`

## Checklist:

- [x] My code follows the **[style guidelines](./CONTRIBUTING.md#style-guides)** of this project.
- [x] I have performed a **self-review** on my own code.
- [x] I have **commented** my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the **[documentation](./CONTRIBUTING.md#documentation-style-guide)**.
- [x] My changes generate **no new warnings**.
- [x] I have added **tests** that prove my fix is effective or that my feature works.
- [x] New and existing **unit tests pass locally** with my changes.

[EMP-3937]: https://searchbroker.atlassian.net/browse/EMP-3937?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/x-components/src/components/no-element.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { defineComponent } from 'vue';
-import { useNoElementRender } from '../composables/index';
+import { useNoElementRender } from '../composables';
```
Comment (diegopf) [inline packages/x-components/src/components/no-element.ts:2]:
```suggestion
import { useNoElementRender } from '../composables/use-no-element-render';
```

File: packages/x-components/src/components/animations/animate-width.vue [hunk 2]
```diff
@@ -1,5 +1,6 @@
 <template>
-  <transition v-on="$listeners" name="x-animate-width-" v-bind="$attrs">
+  <!-- eslint-disable-next-line vue/attributes-order -->
+  <transition v-bind="$attrs" v-on="$listeners" name="x-animate-width-">
```
Comment (diegopf) [inline packages/x-components/src/components/animations/animate-width.vue:3]: I'm just curious, why the attributes order needs to change here?

File: packages/x-components/src/components/animations/fade.vue [hunk 3]
```diff
@@ -1,5 +1,6 @@
 <template>
-  <transition v-on="$listeners" name="x-fade-" v-bind="$attrs" :appear="appear">
+  <!-- eslint-disable-next-line vue/attributes-order -->
+  <transition v-bind="$attrs" v-on="$listeners" name="x-fade-" :appear="appear">
```
Comment (diegopf) [inline packages/x-components/src/components/animations/fade.vue:3]: I'm just curious, why the attributes order needs to change here?

File: packages/x-components/src/components/animations/animate-width.vue (hunk 2 above)
Comment (alvarodE) [reply packages/x-components/src/components/animations/animate-width.vue:3]: While Vue 2 doesn't care about props merge when using v-bind in Vue 3 the order is important: https://v3-migration.vuejs.org/breaking-changes/v-bind

To migrate without a warning we should put the v-bind always first. In this specific case we will also remove the v-on once we migrate to Vue 3 as it is included in the new v-bind: https://v3-migration.vuejs.org/breaking-changes/v-on-native-modifier-removed.html

**Review by diegopf — APPROVED** [review]

**Review by joseacabaneros — APPROVED** [review]

**Pull request merged**